FrEee/Scripts/AI/TestAI/MinistryOfColonization.csx

```diff
@@ -0,0 +1,293 @@
+                var potentialPriority =  plan.Priority - (distance * 100); 
+
+                if (potentialPriority > currentPriority)
+                {
+                    currentPriority = potentialPriority;
+                    currentPlan = plan; 
+                }
+            }
+            if (currentPlan == null)
+                continue; 
+
+            currentPlan.Ship = ship;
+            currentPlan.AwaitingBuild = false;
+            if (currentPlan.ShipBuildPlan != null)
+            {
+                currentPlan.ShipBuildPlan.IsComplete = true;
+            }
+        }
+
+
+
+        var plansToProcess = PlanManager.CurrentPlans.ColonizationPlans.Where(x => !x.ColonyShipDespatched && x.Ship != null); 
+        foreach(var plan in plansToProcess)
+        {
+            ///someone got here first?! Cancel the plan. 
+            if (plan.Planet.Colony != null)
+            {
+                plan.IsComplete = true; 
+                continue; 
+            }
+            var ship = plan.Ship; 
+            //load some population. 
+            if (ship.Cargo.Population.Count < 1)
+            {
+                var planets = ship.Sector.SpaceObjects.OfType<Planet>().Where(x => x.Owner == Empire); 
+                if (planets.Count() > 0)
+                {
+                    bool foundBreathers = false; 
+                    foreach (var pHere in planets)
+                    {
+                        var delta = new CargoDelta();
+                        foreach (var kvp in pHere.AllPopulation)
+                        {
+                            if (kvp.Key.NativeAtmosphere == plan.Planet.Atmosphere)
+                            {
+                                delta.RacePopulation[kvp.Key] = null; // load all population of this race
+                                foundBreathers = true;
+                            }
+                        }
+                        if (foundBreathers)
+                        {
+                            var loadPopOrder = new TransferCargoOrder(true, delta, pHere);
+                            ship.IssueOrder(loadPopOrder);
+                        }
+                    }
+                    if (!foundBreathers)
+                    {
+                        foreach (var pHere in planets)
+                        {
+                            var delta = new CargoDelta();
+                            delta.AllPopulation = true;
+                            var loadPopOrder = new TransferCargoOrder(true, delta, pHere);
+                            ship.IssueOrder(loadPopOrder); 
+                        }
+                    }
+                }
+                else
+                {
+                    //insert code to head to nearest colony? 
+                    continue; 
+                }
+            }
+
+            plan.ColonyCommandId = DateTime.UtcNow.Ticks; 
```

Comment:
Why are you assigning a timestamp as an ID? Why not use `Galaxy.Current.AssignID(plan);` or something like that?